src/fs.rs

```diff
@@ -149,6 +153,8 @@ impl FilesystemReader<'_> {
 
         let mut buffer = [0; 65536];
         for name in names.split_inclusive(|c| *c == 0) {
+            // SAFETY: casting i8 to u8 is safe.
+            let name: &[u8] = unsafe { std::mem::transmute(name) };
```

Comment:
I really wanted to avoid `unsafe` but my old `//fml` solution was causing warnings in the case that `c_char` ended up already being `u8`.  I guess this one won't.

I also expect this to be relatively short-lived until rustix figures out the solution here...